desci-models/src/transformers/RoCrateTransformer.ts

```diff
@@ -111,12 +147,237 @@ export class RoCrateTransformer implements BaseTransformer {
+    let keywordsArray: string[] = [];
+    if (nodeObject.keywords && nodeObject.keywords.length > 0) {
+      keywordsArray = nodeObject.keywords;
+    } else {
+      // Try to extract keywords from component metadata
+      nodeObject.components.forEach((component) => {
+        const payload = component.payload as any;
+        if (payload?.keywords) {
+          if (typeof payload.keywords === 'string') {
+            keywordsArray.push(...payload.keywords.split(',').map((k: string) => k.trim()));
+          } else if (Array.isArray(payload.keywords)) {
+            keywordsArray.push(...payload.keywords);
+          }
+        }
+      });
+      // If still no keywords, add defaults for discoverability
+      if (keywordsArray.length === 0) {
+        keywordsArray = ['research', 'dataset', 'open science', 'FAIR data'];
+      }
+    }
+    // Deduplicate and set keywords
+    keywordsArray = [...new Set(keywordsArray)];
+    const keywordsStr = keywordsArray.join(', ');
+    rootEntity.keywords = keywordsStr;
+    rootEntity['schema:keywords'] = keywordsStr;
+    rootEntity['http://schema.org/keywords'] = keywordsStr; // Full URI
+    rootEntity['dcterms:subject'] = keywordsArray;
+    if (authors && authors.length > 0) {
+      const creatorRefs = authors
+        .filter((a) => a['@id'])
+        .map((a) => ({
+          '@id': a['@id'],
+        }));
+      rootEntity.creator = creatorRefs;
+      rootEntity['schema:creator'] = creatorRefs; // Explicit schema.org
+      rootEntity['dcterms:creator'] = authors.filter((a) => a.name).map((a) => a.name); // Dublin Core (names only)
+    }
+
+    // Build distribution array for FAIR F3.01M (data content identifiers)
+    // F-UJI looks for distribution/hasPart with contentUrl to find data files
+    // Only include components with valid IPFS CIDs, not external links
+    const distribution: any[] = [];
+    const fileSizes = metadata?.fileSizes || {};
+    nodeObject.components.forEach((component) => {
+      // Skip LINK components as they point to external URLs, not IPFS content
+      if (component.type === ResearchObjectComponentType.LINK) {
+        return;
+      }
+      const payload = component.payload as any;
+      const cid = cleanupUrlOrCid(payload?.cid || payload?.url);
+      // Only include if it looks like a valid IPFS CID (starts with Qm or bafy)
+      if (cid && (cid.startsWith('Qm') || cid.startsWith('bafy') || cid.startsWith('bafk'))) {
+        const contentUrl = `${DESCI_IPFS_RESOLVER_HTTP}${cid}`;
+        const downloadEntry: any = {
+          '@type': 'DataDownload',
+          'contentUrl': contentUrl,
+          'url': contentUrl,
+          'name': component.name,
+          'encodingFormat': component.type === ResearchObjectComponentType.PDF ? 'application/pdf' :
+                          component.type === ResearchObjectComponentType.CODE ? 'text/plain' :
+                          'application/octet-stream',
```

Comment:
Hmm, not entirely sure but I think a code component could be a directory as well which might complicate things a bit here :thinking: